### PR TITLE
python3Packages.pytorch-lightning: fix build, add deps

### PR DESCRIPTION
**Pull request by cfhammill**
###### Description of changes

adds missing propagatedBuildInputs
pydeprecate: init at 0.3.1
torchmetrics: init at 0.7.3

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cfhammill** [comment]
cc @bcdarwin, this PR would be up your alley. I should have noted above, the current build for pytorch-lightning is broken and fails due to missing dependencies pyDeprecate, torchmetrics, and fsspec, the first two of those needed to be added to nixpkgs.

**Comment by cfhammill** [comment]
editorconfig error seems spurious

File: pkgs/development/python-modules/pydeprecate/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "pyDeprecate";
```
Comment (FRidh) [inline pkgs/development/python-modules/pydeprecate/default.nix:8]: pname should stay the same as the attribute name, so without uppercase.
In fetchPypi you can use the one with uppercase

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pydeprecate/default.nix (hunk 1 above)
Comment (cfhammill) [reply pkgs/development/python-modules/pydeprecate/default.nix]: thanks for the suggestion, fixed now

**Comment by bcdarwin** [comment]
Fixed merge conflict (tensorflow-tensorboard got renamed to tensorboard).

**Comment by bcdarwin** [comment]
Build is currently failing due to this release specifying an exact point release of `setuptools` -- need to either update `pytorch-lightning` or jailbreak `setuptools` in `requirements.txt`.

**Comment by bcdarwin** [comment]
Mostly addressed by #172300 except for `python3Packages.pydeprecate` addition.

**Pull request closed without merging**